File: Javascript/level 5/Hanoi Tower.js
var answer = [];
function move(n,a,b,c){
    var arr=[];
    if(n==1){
        arr.push(a);
        arr.push(c);
        answer.push(arr);
    }
    else{
        move(n-1,a,c,b);
        arr.push(a);
        arr.push(c);
        answer.push(arr);
        move(n-1,b,a,c);
    }
}
function hanoi(n) {
    answer=[];
    move(n,1,2,3);
	return answer;
	// 2차원 배열을 반환해 주어야 합니다.
}

// 아래는 테스트로 출력해 보기 위한 코드입니다.
console.log(hanoi(2));

File: CPP/level 5/Hanoi Tower.cpp
#include<iostream>
#include<vector>
using namespace std;

vector<vector<int> > answer;
void move(int n, int a, int b, int c){
    vector<int> arr;
    if(n==1){       
        arr.push_back(a);
        arr.push_back(c);
        answer.push_back(arr);
    }
    else{
        move(n-1,a,c,b);
        arr.push_back(a);
        arr.push_back(c);
        answer.push_back(arr);
        move(n-1,b,a,c);
    }
}
vector<vector<int> > hanoi(int no)
{
    move(no,1,2,3);
	return answer;
}
int main()
{
	int testNo = 2;

	vector<vector<int> > testAnswer = hanoi(testNo);

	// 아래는 테스트로 출력해 보기 위한 코드입니다.
	for(int i=0; i< testAnswer.size(); i++)
	{
		for(int j=0; j<testAnswer[i].size(); j++)
		{
			cout << testAnswer[i][j] << " ";
		}
		cout << "\n";
	}
}
